clang-tools-extra/clang-tidy/ClangTidyDiagnosticConsumer.cpp

```diff
@@ -578,6 +579,13 @@ llvm::Regex *ClangTidyDiagnosticConsumer::getHeaderFilter() {
   return HeaderFilter.get();
 }
 
+llvm::Regex *ClangTidyDiagnosticConsumer::getExcludeHeaderFilter() {
+  if (!ExcludeHeaderFilter)
+    ExcludeHeaderFilter = std::make_unique<llvm::Regex>(
+        *Context.getOptions().ExcludeHeaderFilterRegex);
```

Comment:
what if ExcludeHeaderFilterRegex is not set ? This look like legacy bug, same as in getHeaderFilter

Reply:
I'm not sure I follow here. If `ExcludeHeaderFilterRegex` isn't set via the command line or config, it will be initialized to an empty string. Then the `llvm::Regex` will be constructed with that empty string.

Is that different than what you mean?

Reply:
My idea were simply to keep this optional uninitialized in such case, instead of initializing it with empty string, and then simply do not call match for such regexp

Reply:
Ok, thanks for clarifying. I will look into this.

Reply:
I just pushed a commit that omits the check when `ExcludeHeaderFilter` is not set by the user. Please let me know if this aligns with your idea.